src/server/req_modify.c

```diff
@@ -848,90 +911,124 @@ req_modifyReservation(struct batch_request *preq)
 			case RESV_ATR_end:
 				temp = strtol(psatl->al_value, &end, 10);
 				if (!is_standing || temp < next_occr_start) {
-					send_to_scheduler = RESV_END_TIME_MODIFIED;
-					presv->ri_alter_etime = presv->ri_wattr[RESV_ATR_end].at_val.at_long;
-					presv->ri_alter_flags |= RESV_END_TIME_MODIFIED;
+					send_to_scheduler = 1;
+					presv->ri_alter.ra_etime = presv->ri_wattr[RESV_ATR_end].at_val.at_long;
+					presv->ri_alter.ra_flags |= RESV_END_TIME_MODIFIED;
 				} else {
-					resv_revert_alter_times(presv);
-					snprintf(log_buffer, sizeof(log_buffer), "%s", msg_stdg_resv_occr_conflict);
+					resv_revert_alter(presv);
 					log_event(PBSEVENT_RESV, PBS_EVENTCLASS_RESV, LOG_INFO,
-						preq->rq_ind.rq_modify.rq_objname, log_buffer);
+						  preq->rq_ind.rq_modify.rq_objname, msg_stdg_resv_occr_conflict);
 					req_reject(PBSE_STDG_RESV_OCCR_CONFLICT, 0, preq);
 					return;
 				}
 
 				break;
 			case RESV_ATR_duration:
-				send_to_scheduler = RESV_DURATION_MODIFIED;
-				presv->ri_alter_flags |= RESV_DURATION_MODIFIED;
+				send_to_scheduler = 1;
+				presv->ri_alter.ra_flags |= RESV_DURATION_MODIFIED;
+				break;
+			case RESV_ATR_resource:
+				if (strcmp(psatl->al_resc, "select") == 0) {
+					if (presv->ri_wattr[RESV_ATR_state].at_val.at_long != RESV_CONFIRMED && presv->ri_wattr[RESV_ATR_state].at_val.at_long != RESV_DEGRADED) {
```

Comment:
just for readability's sake, can you make this `if (!(a or b))`, personally i think it's cleaner
it would read if not confirmed or degraded, then do this

Reply:
I don't think it is more readable.  I do agree this statement being so long makes it hard to read.  I set the state in a local variable and used it in the if statement.  I think it is much easier to read now.